src/main-process/atom-application.coffee

```diff
@@ -67,7 +67,7 @@ class AtomApplication
     {@resourcePath, @devResourcePath, @version, @devMode, @safeMode, @socketPath, @logFile, @userDataDir} = options
     @socketPath = null if options.test or options.benchmark or options.benchmarkTest
     @pidsToOpenWindows = {}
-    @windows = []
+    @windows = new WindowStack()
```

Comment:
🎨  Might be nice to rename `@windows` to `@windowStack` to help emphasize that it's an object, not an array.